### DISCUSSION TITLE
VITS adversarial necessity and EATS paper lessons learnt.

### DISCUSSION THREAD

**Discussion by p0p4k** [coqui-ai/TTS#1506] [General]
As you know, in my previous discussion about VITS paper #1501 , I was wondering why is there an adversarial training despite the variational loss comparing the mel-spectrograms directly. 
The authors of VITS cited the [EATS paper](https://arxiv.org/abs/2006.03575) while mentioning adversarial training which i missed earlier. 
Now, since I did not read EATS paper earlier, my formulation of the question was in reverse order. From EATS paper, the authors use adversarial first and then propose using mel loss. In section 2.4 (EATS), they say that a conditional generator (conditioned on text, producing waveform), can work in theory, but in practice, the training gets stuck (because initially the alignment might be really poor, never helping the model to learn anything). **So, they use an additional mel-spec to kick start the training using the L1 loss.** 
While EATS paper is good, the results from VITS are better. I will explore more of these papers and try to summarise their math in my upcoming discussion posts.  I hope my understanding of these papers has been correct so far, if not please let me know. Thanks.

Additional [video resource from EATS author](https://crossminds.ai/video/eats-end-to-end-adversarial-text-to-speech-60c3d2566af07cfaf7325dca/) explaining "chicken-egg" problem, of why the conditional adversarial training (conditioned on text) didnt work well at ~4:00 into the video .

**Comment by erogol** [comment]
If I am not mistaken, the adversarial loss is only needed for the vocoder part of the VITS model which is reminiscent of the HiFiGAN vocoder. 

Variational loss is only necessary to learn the features that the vocoder part is conditioned on.  

Even though what EATS proposed makes sense, it does not hold for the VITS training. I'd claim it is because VITS is really quick to learn the durations and the output features due to its probabilistic nature. Thus it is faster in providing a helpful signal to the vocoder component.

**Reply by p0p4k** [reply]
I see, Ill read HiFiGAN and try to understand this better. Thanks!!!
